Movie Challenge/Movie Challenge/Controllers/Detail/DetailViewController.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  NewDetailViewControll.swift
+//  Movie Challenge
+//
+//  Created by igor gomes arantes on 05/09/18.
+//  Copyright © 2018 igor gomes arantes. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class DetailViewController : UIViewController, UICollectionViewDelegate, UICollectionViewDataSource{
+    
+    private var movie = MovieDTO()
+    private var imageDownloadTask: URLSessionDataTask!
+    private var movieId: Int!
+    private var favorite: Bool!
+
+    @IBOutlet weak var overviewLabelView: UILabel!
+    @IBOutlet weak var numberOfVotesLabelView: UILabel!
+    @IBOutlet weak var pointsLabelView: UILabel!
+    @IBOutlet weak var runtimeLabelView: UILabel!
+    @IBOutlet weak var yearLabelView: UILabel!
+    @IBOutlet weak var titleLabelView: UILabel!
+    @IBOutlet weak var posterImageView: UIImageView!
+    @IBOutlet weak var categoryCollectionView: UICollectionView!
+    @IBOutlet weak var favoriteButtonView: UIButton!
+    @IBOutlet weak var categoryCollectionViewHeightConstraint: NSLayoutConstraint!
+    
+    @IBAction func favoriteMovie(_ sender: Any) {
+        if(favorite){
+            do{
+                try MovieRepository.shared().remove(id: movieId)
+                favorite = false
+                setButtonState()
+                
+                print("Removeu")
+            }catch let error{
+                print("Erro ao deletar o filme", error)
+            }
+            
+        }else{
+            do{
+                try MovieRepository.shared().save(movie: movie)
+                favorite = true
+                setButtonState()
+                
+                print("Salvou")
+            }catch let error{
+                print("Erro ao salvar o filme", error)
+            }
+        }
+    }
+    
+    private func setPoster(poster: UIImage){
+        DispatchQueue.main.async() {
+            self.posterImageView.image = poster
+            self.posterImageView.setNeedsDisplay()
+        }
+    }
+    
+    private func setButtonState(){
+        if(favorite){
+            DispatchQueue.main.async(){
+                self.favoriteButtonView.backgroundColor = #colorLiteral(red: 0.9254902005, green: 0.2352941185, blue: 0.1019607857, alpha: 1)
+                self.favoriteButtonView.setTitle("Remover", for: UIControlState.normal)
+            }
+        }else{
+            DispatchQueue.main.async(){
+                self.favoriteButtonView.backgroundColor = #colorLiteral(red: 1, green: 0.6116010603, blue: 0.006196474039, alpha: 1)
+                self.favoriteButtonView.setTitle("Favoritar", for: UIControlState.normal)
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
```

Comment:
To organize better your code, put the primitives methods first in your the class.